generator/parse.py

```diff
@@ -272,7 +358,11 @@ def parse(self, l, ctxt):
             mm = re.match(r'\s*((?:[\w:_]|< ?| ?>| ?, ?)+)(?<!return|delete)\s+[\*&]?([\w_]+)\s*;', l)
             if mm :
                 if mm.group(2) not in ctxt.gdict['classes'][ctxt.curr_class].members:
-                    ctxt.gdict['classes'][ctxt.curr_class].members[mm.group(2)] = mm.group(1)
+                    # if multiline documentation is found then save the destination for documentation
+                    if ctxt.doc_state is doc_state.multi:
+                        ctxt.doc_lambda = lambda: ctxt.gdict['classes'][ctxt.curr_class].members[mm.group(2)]
+                    ctxt.gdict['classes'][ctxt.curr_class].members[mm.group(2)] = [mm.group(1), ctxt.doc]
+                    ctxt.doc = ''
                 return True
```

Comment:
what about always defining the lambda and putting the rest of the logic into an extra handler which is called at the end of the parser-loop below?